### PR TITLE
Find user and users by using employee ID

### LINKED ISSUE #1120: Find users based on employee_id
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
We need to get the user or users_list based on the employee_id in azure ad

### New or Affected Resource(s)

* azuread_user
* azuread_users

### Potential Terraform Configuration

```hcl
data "azuread_user" "user" {
  employee_id = "123"
}
```
```hcl
data "azuread_users" "users" {
  employee_id = ["123", "125"]
}
```

### References

* #1040

### PR DESCRIPTION
**Pull request by pchanvallon**
Add the possibility to `azuread_user` and `azuread_users` data sources to search Azure AD users using `employee_id` property.

closes #1120

### CODE REVIEW

**Comment by pchanvallon** [comment]
Hello @manicminer, 
Is it possible to have a review on this PR ? 
I'm really looking for using it on my various projects ? 
Thanks a lot.

File: docs/data-sources/users.md [hunk 1]
```diff
@@ -26,18 +26,20 @@ data "azuread_users" "users" {
...
 
+* `employee_ids` - (Optional) The employee identifiers assigned to the users by the organisation.
 * `ignore_missing` - (Optional) Ignore missing users and return users that were found. The data source will still fail if no users are found. Cannot be specified with `return_all`. Defaults to `false`.
 * `mail_nicknames` - (Optional) The email aliases of the users.
 * `object_ids` - (Optional) The object IDs of the users.
 * `return_all` - (Optional) When `true`, the data source will return all users. Cannot be used with `ignore_missing`. Defaults to `false`.
 * `user_principal_names` - (Optional) The user principal names (UPNs) of the users.
 
-~> Either `return_all`, or one of `user_principal_names`, `object_ids` or `mail_nicknames` must be specified. These _may_ be specified as an empty list, in which case no results will be returned.
+~> Either `return_all`, or one of `user_principal_names`, `object_ids`, `mail_nicknames` or `employee_id` must be specified. These _may_ be specified as an empty list, in which case no results will be returned.
```
Comment (manicminer) [inline docs/data-sources/users.md:36]:
```suggestion
~> Either `return_all`, or one of `user_principal_names`, `object_ids`, `mail_nicknames` or `employee_ids` must be specified. These _may_ be specified as an empty list, in which case no results will be returned.
```

File: docs/data-sources/users.md [hunk 2]
```diff
@@ -26,18 +26,20 @@ data "azuread_users" "users" {
...
 
 ## Attributes Reference
 
 The following attributes are exported:
 
+* `employee_ids` - The employee identifier assigned to the users by the organisation.
```
Comment (manicminer) [inline docs/data-sources/users.md:42]:
```suggestion
* `employee_ids` - The employee identifiers assigned to the users by the organisation.
```

**Review by manicminer — APPROVED** [review]
Many thanks for this contribution @pchanvallon, this LGTM 👍

**Comment by manicminer** [comment]
Test results

Screenshot 2023-07-13 at 22 54 17

File: internal/services/users/user_data_source.go [hunk 3]
```diff
@@ -26,11 +26,20 @@ func userDataSource() *schema.Resource {
 		},
 
 		Schema: map[string]*schema.Schema{
+			"employee_id": {
+				Description:      "The employee identifier assigned to the user by the organisation",
+				Type:             schema.TypeList,
```
Comment (manicminer) [inline internal/services/users/user_data_source.go:31]:
```suggestion
				Type:             schema.TypeString,
```

File: internal/services/users/user_data_source_test.go [hunk 4]
```diff
@@ -198,3 +217,25 @@ data "azuread_user" "test" {
 }
 `, data.RandomInteger)
 }
+
+func (UserDataSource) byEmployeeId(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%[1]s
+
+data "azuread_user" "test" {
+	employee_id = azuread_user.test.employee_id
```
Comment (manicminer) [inline internal/services/users/user_data_source_test.go:226]:
```suggestion
  employee_id = azuread_user.test.employee_id
```

**Comment by manicminer** [comment]
Test results

Screenshot 2023-07-14 at 00 17 32

**Pull request merged**